modules/captifyRtdProvider.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * This module adds Captify real time data provider module
+ * The {@link module:modules/realTimeData} module is required
+ * The module will fetch segments (page-centric) from Captify live-classification server
+ * @module modules/captifyRtdProvider
+ * @requires module:modules/realTimeData
+ */
+import { submodule } from '../src/hook.js';
+import { getRefererInfo } from '../src/refererDetection.js';
+import { ajax } from '../src/ajax.js';
+import { config } from '../src/config.js';
+import {deepAccess, isArray, isEmptyStr, isNumber, logError} from '../src/utils.js';
+
+const MODULE_NAME = 'realTimeData';
+const SUBMODULE_NAME = 'CaptifyRTDModule';
+const DEFAULT_LC_URL = 'https://live-classification.cpx.to/prebid-segments';
+
+const STATUS = {
+  SUCCESS: 200,
+  ACCEPTED: 202,
+};
+
+/**
+ * Set `appnexusAuctionKeywords` that appnexus bidder will read and send in request to Xandr
+ * @param {Array<number>} segments captify segments for Appnexus(Xandr) system, in form of [id1, id2, id3]
+ *                 where id1, id2, id3 - actual Xandr segment ids with keywords enabled
+ */
+export function setAppnexusSegments(segments) {
+  config.setConfig({
+    appnexusAuctionKeywords: {
+      'captify_segments': segments,
+    },
+  })
+}
+
+/**
+ * Function returns only bidders that contained both, in moduleConfig and at least one adUnit.
+ * @param {Array<string>} bidders Contains list of bidders, to set targeting for
+ * @param {Object} data Response of live-classification service
+ */
+export function addSegmentData(bidders, data) {
+  for (const bidder of bidders) {
+    if (bidder === 'appnexus') { setAppnexusSegments(data['xandr']) }
+  }
+}
+
+/**
+ * Function returns only bidders that contained in both, moduleConfig and at least one adUnit.
+ * @param {Object} moduleConfig Config object passed to the module
+ * @param {Object} reqBidsConfigObj Config object for the bidders; each adapter has its own entry
+ */
+export function getMatchingBidders(moduleConfig, reqBidsConfigObj) {
+  const biddersFromConf = deepAccess(moduleConfig, 'params.bidders');
+
+  const adUnitBidders = reqBidsConfigObj.adUnits
+    .flatMap(({bids}) => bids.map(({bidder}) => bidder))
+    .filter((e, i, a) => a.indexOf(e) === i);
+
+  if (!isArray(adUnitBidders) || !adUnitBidders.length) {
+    throw new Error('Missing parameter bidders in bidRequestConfig');
```

Comment:
I think it's better to use logError function instead of throw new Error, and not catching it. You're already using logError in line 112, maybe use that?